### PR TITLE
Add plugin sample

### PR DESCRIPTION
**Pull request by virzak**

### CODE REVIEW

File: samples/core/SqlServer/Plugin/PluginSetup.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+        public override void PopulateDebugInfo(IDictionary<string, string> debugInfo)
+            => debugInfo["SqlServer:" + nameof(SqlServerExtensions.UseAugmentation)] = "1";
+
+        public override string LogFragment
+            => "using Augmentation";
+    }
+}
+
+public static class SqlServerNetTopologySuiteServiceCollectionExtensions
```
Comment (roji) [inline samples/core/SqlServer/Plugin/PluginSetup.cs:47]: The type name seems wrong

File: samples/core/SqlServer/Plugin/DbFunctionsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+
+namespace SqlServer.Plugin;
+public static class DbFunctionsExtensions
```
Comment (roji) [inline samples/core/SqlServer/Plugin/DbFunctionsExtensions.cs:6]: The class should be named something like AugmentationDbFunctionsExtensions

**Review by roji — COMMENTED** [review]
Thanks for this PR.

We have plans to improve documentation on writing providers (#681, for [this page](https://docs.microsoft.com/en-us/ef/core/providers/writing-a-provider)), we could also have a page on how to write a plugin.

We usually add samples with the corresponding conceptual documentation; a code sample alone (such as this) isn't easily discoverable by users looking at the docs. If the goal is simply to have a working plugin source to look at, then simply pointing to e.g. the NetTopologySuite source code may be enough.

Note also that  this sample shows only IMethodCallTranslatorPlugin, but there are other plugin types (e.g. ITypeMappingSourcePlugin, IEvaluatableExpressionFilterPlugin, IConventionSetPlugin...).

File: samples/core/SqlServer/Plugin/PluginSetup.cs (hunk 1 above)
Comment (virzak) [reply samples/core/SqlServer/Plugin/PluginSetup.cs]: Fixed

File: samples/core/SqlServer/Plugin/DbFunctionsExtensions.cs (hunk 2 above)
Comment (virzak) [reply samples/core/SqlServer/Plugin/DbFunctionsExtensions.cs:6]: Fixed

**Comment by virzak** [comment]
Hi @roji,

This PR was started in an attempt to reproduce a possible bug in EF Core PawelGerr/Thinktecture.EntityFrameworkCore#17

I didn't find an existing test case to modify in [dotnet/efcore](https://github.com/dotnet/efcore), because I'm either not familiar with it or because it wasn't there.

Still hoping to reproduce the issue, but in the meantime the PR was a good exercise for me to isolate the plugin concept. If the PR gets merged and helps someone else - that's a bonus.

**Comment by virzak** [comment]
Issue dotnet/efcore#27075 was the original reason for this PR

**Comment by roji** [comment]
@virzak we've discussed this internally, and the decision is that we prefer not to add standalone code samples that aren't part of corresponding conceptual documentation. In other words, something like this would definitely make sense as part of #681, but we don't think that having the code without any docs is very helpful.

Good luck with your bug and let us know if you need any help.

**Comment by virzak** [comment]
@roji Yeah, that's cool. You mentioned that you're pointing developers to NetTopologySuite source code. Perhaps a plugin/README.md could have that link until #681 is complete. It would make sense for me, as I am who tends to look at the source code a lot more than reading documentation.

**Comment by roji** [comment]
I'm not sure a plugin/README.md would be very discoverable either - I think at some point we need to do #681...

**Pull request closed without merging**